mlir-sycl/include/mlir/Dialect/SYCL/Transforms/Passes.td

```diff
@@ -11,6 +11,23 @@
 
 include "mlir/Pass/PassBase.td"
 
+def ConstantPropagationPass : Pass<"sycl-constant-propagation", "gpu::GPUFuncOp"> {
+  let summary = "SYCL host-device constant propagation";
+  let description = [{
+    This pass implements SYCL host-device constant propagation. In order to do
+    so, it tracks constant values, e.g., those defined by `arith.constant`,
+    passed as an argument to `sycl.host.schedule_kernel`, drops them in host
+    side and replaces the corresponding `gpu.func` argument with the constant
+    value, also dropping the argument in device side.
```

Comment:
Given that we are going to drop the host code, would changing number of arguments affect correctness?

Reply:
No, this is handled by the SYCL RT. Unused kernel arguments will be marked as such in the kernel argument usage mask, which is part of the device binary. The SYCL RT decodes this mask and does pass an argument for arguments marked as unused. 

As this is completely handled by the combination of device code and SYCL RT, dropping the host code is still fine.

Reply:
I won't drop the host code to avoid signature mismatch